pkg/apis/provisioning/v1alpha1/provisioner.go

```diff
@@ -21,9 +21,32 @@ import (
 
 type ProvisionerSpec struct {
 	// +optional
-	Cluster      *ClusterSpec     `json:"cluster,omitempty"`
-	Allocation   AllocationSpec   `json:"allocation,omitempty"`
-	Reallocation ReallocationSpec `json:"reallocation,omitempty"`
+	Cluster *ClusterSpec `json:"cluster,omitempty"`
+	// Taints will be applied to every node launched by the Provisioner If
+	// specified, the provisioner will not be able to provision nodes for pods
+	// that do not have matching tolerations.
+	// +optional
+	Taints []v1.Taint `json:"taints,omitempty"`
+	// Labels will be applied to every node launched by the Provisioner unless
+	// overriden by pod node selectors. Well known labels control provisioning
+	// behavior. Additional labels may be supported by your cloud
+	// provider.
+	// Well known, supported labels:
+	//      "kubernetes.io/arch"
+	//      "kubernetes.io/os"
+	//      "topology.kubernetes.io/zone"
+	//      "node.kubernetes.io/instance-type"
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+	// Zones constrains where nodes will be launched by the Provisioner. If
+	// unspecified, defaults to all zones in the region. Cannot be specified if
+	// label "topology.kubernetes.io/zone" is specified.
```

Comment:
I need to document this, and we probably need to discuss, but here's what I was thinking.

pod.spec.nodeSelector["topology.kubernetes.io/zone"] <-- Override, the pod will *always* schedule here
provisioner.spec.labels["topology.kubernetes.io/zone"]  <-- Default (single value), mostly provided for consistency. Alternatively we can reject/reserve this keyword in favor of the next option
provisioner.spec.Zones <-- Default (multiple values), this first class concept is necessary to allow restricting to multiple zone options. We will likely recommend this one to users, but support the single specification way as it fits so cleanly into the conceptual model.

Thoughts?

Reply:
My very quick take is that 1 & 3 make sense, it would be nice to drop #2 (just because then there would be "two ways to do it"); do we think people will want to use labels on the provisioner to control this?

Reply:
This decision means that we need to treat "topology.kubernetes.io/zone" as reserved and prevent users from creating provisioners with this label. Agree w.r.t avoiding "two ways to do it".